packages/SwingSet/src/controller/controller.js

```diff
@@ -303,22 +311,14 @@ export async function makeSwingsetController(
    * @returns { Promise<BundleID> }
    */
   async function validateAndInstallBundle(bundle, allegedBundleID) {
-    // TODO: validation: unpack, parse sources, check hashes
-
-    // during the transition to endo's new format, preemptively ignore the
-    // hash it provides
-    bundle = harden({
-      moduleFormat: bundle.moduleFormat,
-      endoZipBase64: bundle.endoZipBase64,
-    });
-
-    // this only computes the hash of the compartment map, it does not check
-    // that the rest of the bundle matches
-    const bundleID = await computeBundleID(bundle);
-    if (allegedBundleID) {
+    await checkBundle(bundle, computeSha512);
+    const { endoZipBase64Sha512 } = bundle;
+    assert.typeof(endoZipBase64Sha512, 'string');
```

Comment:
This assert is necessary because the bundle could be internally self-consistent, but be of a format that doesn't provide `endoZipBase64Sha512`, right? Just making sure I understand that that's the only surprise that could make it past `checkBundle`.

Reply:
I should also verify `moduleFormat`. At the moment, `checkBundle` will only bless a bundle if `moduleFormat` is `"endoZipBase64"`, `endoZipBase64` is a string, and `endoZipBase64Sha512` is a string. So this is fine for now, but in the face of evolution, `checkBundle` should eventually allow other formats.

I’m less concerned here about surprises coming out of `checkBundle` and more about other ways bundles might arrive here that might not necessarily get checked. The presence of this property is necessary to compute an ID for now.